synapse/storage/data_stores/main/e2e_room_keys.py

```diff
@@ -162,6 +152,95 @@ def get_e2e_room_keys(self, user_id, version, room_id=None, session_id=None):
 
         return sessions
 
+    def get_e2e_room_keys_multi(self, user_id, version, room_keys):
+        """Get multiple room keys at a time.  The difference between this function and
+        get_e2e_room_keys is that this function can be used to retrieve
+        multiple specific keys at a time, whereas get_e2e_room_keys is used for
+        getting all the keys in a backup version, all the keys for a room, or a
+        specific key.
+
+        Args:
+            user_id(str): the user whose backup we're querying
+            version(str): the version ID of the backup we're querying about
+            room_keys(dict[dict[iterable[str]]]): a map of room IDs to dict which
```

Comment:
the type syntax includes the key type: `dict[str, dict[str, iterable[str]]]`

We can maybe simplify the description:

```
            room_keys (dict[str, dict[str, iterable[str]]]): 
                map from roomID -> {"session": [session ids]}
```